File: src/main/java/com/oshi/libgenericsettings/data/IconTitleData.java
package com.oshi.libgenericsettings.data;

/**
 * Settings Icon & Title data object. Being use in layout/view_type_icon_title.xml
 */
public class IconTitleData extends TitleData {

    private int iconResId;


    public IconTitleData(String title) {
        super(title);
    }

    public IconTitleData(String title, int titleColor) {
        super(title, titleColor);
    }

    public IconTitleData(int iconResId, String title, int titleColor) {
        super(title, titleColor);
        this.iconResId = iconResId;
    }

    public int getIconResId() {
        return iconResId;
    }

    public void setIconResId(int iconResId) {
        this.iconResId = iconResId;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE_SETTINGS_ICON_TITLE;
    }
}

File: src/main/java/com/oshi/libgenericsettings/data/IconTitleData.kt
package com.oshi.libgenericsettings.data

import android.support.annotation.ColorRes
import android.support.annotation.DrawableRes
import com.oshi.libgenericsettings.data.BaseViewTypeData.Companion.SettingsViewTypes

/**
 * Settings Icon & Title data object. Being use in layout/view_type_icon_title.xml
 */
open class IconTitleData(@DrawableRes var iconResId : Int, title : String) : TitleData(title) {

    @ColorRes var iconColor: Int = 0

    override fun getViewType(): SettingsViewTypes {
        return SettingsViewTypes.VIEW_TYPE_SETTINGS_ICON_TITLE
    }
}